sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -327,8 +383,10 @@ int main()
     return rc;
   }
 
-  printf("Started registration. [Press ENTER to abort]\n");
-  (void)getchar();
+  if ((rc = get_operation_status()) != 0)
+  {
+    return rc;
+  }
```

Comment:
We should try to do clean shutdown. 
E.g: If you do not do clean shutdown when possible, you are hogging server side resources (at least by not sending RST packet),

Note: Every exit of app can be turn into bad shutdown, but that does not mean we should not try when possible.

Reply:
You had this comment in the previous PR. In general it's good to fail fast and avoid corrupting further state. 
The sample does exist gracefully if the registration operation completes in either success or failure.